File: Planetary-Formation-main/planetaryFormation.java
import Storage.Positionals.GeoCoord;
import Storage.Positionals.Vector3;
import Storage.STL.TriangleFace;

import java.util.Arrays;
import java.util.Scanner;

public class planetaryFormation {
  public static void main(String[] args) {
    TerrestrialPlanet planet = new TerrestrialPlanet(5, 10000000000L);
    planet.initFractalNoise(5, 6);
    System.out.println(planet.terrain.getPoint(new GeoCoord(Math.toRadians(40), Math.toRadians(20))));
    System.out.println(planet.terrain.getPoint(new GeoCoord(Math.toRadians(40), Math.toRadians(21))));
    System.out.println(planet.terrain.getPoint(new GeoCoord(Math.toRadians(90), Math.toRadians(128))));

    System.out.println(Arrays.toString(TriangleFace.fromOrientingPoint(new Vector3(new double[]{0, 0, 0}), true, new Vector3[]{new Vector3(new double[]{9, 4, 5}), new Vector3(new double[]{9, 3, 8}), new Vector3(new double[]{16, 49, 1})}).exportCode()));
  }

  static double getDouble(String prompt, Scanner sc) {
    double radius;
    while (true) {
      try {
        System.out.print(prompt);
        radius = Double.parseDouble(sc.nextLine());
        break;
      } catch(NumberFormatException e) {
        System.out.println("\nNot a valid input. Please input a double.");
      }
    }
    return radius;
  }
}

File: Planetary-Formation-main/PlanetaryFormation.kt
import storage.PlanarScalarGrid
import storage.ScalarQuadTree
import storage.positionals.GeoCoord
import storage.positionals.Index
import java.io.IOException
import java.util.*
import kotlin.math.absoluteValue

fun main(args: Array<String>) {
    val planet = TerrestrialPlanet(4, 10000000000L)
    //planet.initFractalNoise(1, 6, 1000000.0)
    planet
    println(planet.terrain.getPoint(GeoCoord(Math.toRadians(40.0), Math.toRadians(20.0))))
    println(planet.terrain.getPoint(GeoCoord(Math.toRadians(40.0), Math.toRadians(23.0))))
    println(planet.terrain.getPoint(GeoCoord(Math.toRadians(90.0), Math.toRadians(128.0))))
//    println(TriangleFace.fromOrientingPoint(Vector3(doubleArrayOf(0.0, 0.0, 0.0)), true, arrayOf(Vector3(doubleArrayOf(9.0, 4.0, 5.0)), Vector3(doubleArrayOf(9.0, 3.0, 8.0)), Vector3(doubleArrayOf(16.0, 49.0, 1.0)))).exportCode().contentToString())
//    planet.terrain.exportFaceSTL("../face.stl")

}

fun getDouble(prompt: String, sc: Scanner): Double {
    var radius: Double
    while (true) {
        try {
            print(prompt)
            radius = sc.nextLine().toDouble()
            break
        } catch (e: NumberFormatException) {
            println("\nNot a valid input. Please input a double.")
        }
    }
    return radius
}
